### PR TITLE
Add loader to yearly budget

### PR DESCRIPTION
**Pull request by hdiniz**
Fix: https://github.com/opencollective/opencollective/issues/3182

### CODE REVIEW

**Comment by znarf** [comment]
Related: https://github.com/opencollective/opencollective/issues/6195

File: server/graphql/v1/TransactionInterface.js [hunk 1]
```diff
@@ -236,85 +236,64 @@ const TransactionFields = () => {
...
-          }
+
+        const fromCollective = await req.loaders.Collective.byId.load(transaction.FromCollectiveId);
+        if (fromCollective.HostCollectiveId) {
+          return req.loaders.Collective.byId.load(fromCollective.HostCollectiveId);
+        }
+
+        const collective = await req.loaders.Collective.byId.load(transaction.CollectiveId);
+        if (collective.HostCollectiveId) {
+          return req.loaders.Collective.byId.load(fromCollective.HostCollectiveId);
```
Comment (znarf) [inline server/graphql/v1/TransactionInterface.js:251]: Looks unrelated, different PR?

File: server/lib/budget.js [hunk 2]
```diff
@@ -695,59 +695,109 @@ export async function sumCollectivesTransactions(
...
+              SUM(coalesce(t."hostFeeInHostCurrency", 0)) +
+              SUM(coalesce(t."paymentProcessorFeeInHostCurrency", 0)) +
+              SUM(coalesce(t."taxAmount" * t."hostCurrencyFxRate", 0)) "amount"
           FROM "Transactions" t
-          LEFT JOIN "Orders" d ON t."OrderId" = d.id
-          LEFT JOIN "Subscriptions" s ON d."SubscriptionId" = s.id
-          WHERE t."CollectiveId" = :CollectiveId
-          AND t."RefundTransactionId" IS NULL
+          LEFT JOIN "Orders" o ON t."OrderId" = o.id
+          LEFT JOIN "Subscriptions" s ON s.id = o."SubscriptionId"
```
Comment (znarf) [inline server/lib/budget.js:732]: If we want to fetch one time contributions, no need to join Subscriptions.

If we do, it's different than what is implied by "last_twelve_months_one_time_contributions".

Comment (hdiniz) [reply server/lib/budget.js:732]: I think i picked a bad alias for that query. The original one does join with subscriptions, it sums up past recurring contributions that are not active anymore.

File: migrations/20221220120527-add-transaction-subscription-indexes.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.addIndex('Transactions', ['CollectiveId', 'createdAt', 'type']);
```
Comment (Betree) [inline migrations/20221220120527-add-transaction-subscription-indexes.js:6]: Should we remove the existing `CollectiveId-type` index in favor of this one?

Comment (znarf) [reply migrations/20221220120527-add-transaction-subscription-indexes.js:6]: @hdiniz ?

Comment (hdiniz) [reply migrations/20221220120527-add-transaction-subscription-indexes.js:6]: `CollectiveId-Type` is one of the most used indexes on the transactions table, I would look at that on a separate issue to see the query impact. 

This new index might not be compatible with `"CollectiveId-type" btree ("CollectiveId", type)` as it will be ordered by the createdAt before type.

```
select indexrelname, idx_scan, pg_size_pretty(pg_relation_size(indexrelid)) as "Index Size" from pg_stat_all_indexes where schemaname = 'public' and relname = 'Transactions' order by idx_scan desc;
                   indexrelname                   | idx_scan  | Index Size
--------------------------------------------------+-----------+------------
 CollectiveId-FromCollectiveId-type               | 751364426 | 61 MB
 transactions__using_gift_card_from_collective_id | 654061807 | 42 MB
 CollectiveId-type                                | 471101054 | 50 MB
```

Comment (znarf) [reply migrations/20221220120527-add-transaction-subscription-indexes.js:6]: Let's merge like this then.
